src/asciidoc/core-beans.adoc

```diff
@@ -4562,14 +4562,16 @@ automatically resolved, with no special setup necessary.
 	}
 ----
 
-[NOTE]
+[WARNING]
 ====
-`@Autowired`, `@Inject`, `@Resource`, and `@Value` annotations are handled by Spring
+`@Autowired`, `@Inject`, `@Resource`, and `@Value`  annotations are handled by Spring
 `BeanPostProcessor` implementations which in turn means that you __cannot__ apply these
 annotations within your own `BeanPostProcessor` or `BeanFactoryPostProcessor` types (if
 any). These types must be 'wired up' explicitly via XML or using a Spring `@Bean` method.
-====
 
+Also, do not use `@Autowired`, `@Inject`, and `@Resource` annotations within a BeanFactory class nor any of its dependencies. Doing so can result in a `BeanCurrentlyInCreationException`. The `@Value` annotation may also cause the problem, d
+epending on whether it causes a bean to be instantiated.
+====
```

Comment:
I thought about making it into a separate warning. That would have resulted in two adjacent warnings that addressed the same set of annotations, so I combined them. I can split them, though, if that works better.

I didn't have the problem. Someone else did. Then the topic came up on Slack, where various folks mentioned that they had tried this and had trouble. I thought it rated a documentation update.

Several people have tried to use those annotations within a BeanFactory (or the dependencies of a BeanFactory) and have had trouble as a result. You can see the evidence of it in the following issues:
https://jira.spring.io/browse/SPR-9255
https://jira.spring.io/browse/SPR-8704
https://jira.spring.io/browse/SPR-10349

If you can tell me additional details around using @Value in this context, please do. I'll be happy to add them to the document.